# DGRNS: A deep learning framework for gene regulatory network inference from single-cell transcriptomic data
Inferring gene regulatory networks (GRNs) based on gene expression profiles is able to provide an insight into a number of cellular phenotypes from the genomic level and reveal the essential laws underlying various life phenomena. Different from the bulk expression data, single-cell tran-scriptomic data embody cell-to-cell variance and diverse biological information, such as tissue characteristics, transformation of cell types, etc. Inferring GRNs based on such data offers unprec-edented advantages for making a profound study of cell phenotypes, revealing gene functions and exploring potential interactions. However, the high sparsity, noise and dropout events of single-cell transcriptomic data pose new challenges for regulation identification.
We develop a deep learning framework for gene regulatory network inference from sin-gle-cell transcriptomic data, DGRNS, which encodes the row data and combines the advantages of RNN and CNN to train a model capable of distinguishing related gene pairs from unrelated gene pairs. To overcome the limitations of such datasets, it applies sliding windows to extract valuable features while preserving the direction of regulation. DGRNS is constructed as a deep learning model containing GRU network for exploring time-dependent information and CNN for learning spatially related information. Our comprehensive and detailed comparative analysis on the dataset of mouse hematopoietic stem cells illustrates that DGRNS outperforms state-of-the-art methods. The networks inferred by DGRNS are about 10% higher than the AUROC of other unsupervised methods, and 8% higher than the AUPR of other supervised methods. By comparing the predic-tions with the standard network, we discover a number of novel interactions which are proved to be true in some specific cell types. Importantly, DGRNS identifies a series of regulatory relationships with high confidence and functional consistency, but they have not yet been experimentally con-firmed and merit further research.

![image](https://user-images.githubusercontent.com/28176452/120603055-2f18a600-c47e-11eb-9af6-3b2bd258d7be.png)

### Improtant parameter annotation
1. Run gen_genePairList.py to generate gene pair list file from gene expression file and gold network, the output is named as XXX-GPL.csv<br>
(1) gene_expression_path: file path of gene expression profiles. (ExpressionDataOrdered.csv means the samples in expression file have been ordered by pseudotime, as described in paper.<br>
(2) gene_order_path: file path of gene ordering.<br>
(3) gold_network_path: file path of gold network after deleting repeat gene regulations.<br>
(4) FGN_file: the filtered gold network which only include genes in GPL_file.<br>
(5) GPL_file: gene pair list file from gene expression file and gold network with fixed number of genes (500 or 1000 in paper), the output is named as XXX-GPL.csv.<br>
(6) Rank_num: the fixed gene number, 500 or 1000 in paper.<br>
2. Run gen_matrix.py or gen_fullmatrix_TF_.py to generate correlation vectors from gene pair list and gene expression profiles.<br>
(1) gene_pair_list_path: file path of GPL_file, which is generated from gen_genePairList.py.<br>
(2) gene_expression_path: the same as 'gene_expression_path' in gen_genePairList.py.<br>
(3) save_dir: file path of generated correlation vectors.<br>
3. Run gen_datasetSplit.py to split dataset into training set, validation set and test set.<br>
4. Run DGRNS_train.py to train DGRNS, predict test dataset, plot ROC and PRC.<br>
(1) data_path: file path of correlation vectors generated by gen_matrix.py.<br>
(2) model_save_dir: the path of trained DGRNS model.<br>
5. (Optional) Run predict_fullnetwork.py to construct gene regulation relationships of whole dataset, including training, validation and test dataset.<br>
(1) matrix_path: file path of correlation vectors.<br>
(2) model_path: file path of trained DGRNS model, normally the same as 'model_save_dir' in DGRNS_train.py.<br>
(3) save_dir: the path of predictions.<br>
